src/healthcheck.c

```diff
@@ -0,0 +1,422 @@
+	return true;
+}
+
+/* Stop healthcheck timer */
+void healthcheck_timer_stop(healthcheck_timer_t *timer)
+{
+	if (timer == NULL || !timer->timer_active) {
+		return;
+	}
+
+	timer->timer_active = false;
+	timer->status = HEALTHCHECK_NONE;
+
+	/* Remove the GLib timeout source */
+	if (timer->timer_id != 0) {
+		g_source_remove(timer->timer_id);
+		timer->timer_id = 0;
+	}
+}
+
+/* Execute healthcheck command inside container using runtime */
+bool healthcheck_execute_command(const healthcheck_config_t *config, const char *container_id, const char *runtime_path, int *exit_code)
+{
+	if (config == NULL || config->test == NULL || container_id == NULL || runtime_path == NULL || exit_code == NULL) {
+		return false;
+	}
+
+	/* Initialize exit code to failure */
+	*exit_code = -1;
+
+	/* Create stderr pipe to capture error output */
+	int stderr_pipe[2];
+	if (pipe(stderr_pipe) == -1) {
+		nwarnf("Failed to create pipe for healthcheck stderr: %s", strerror(errno));
+		return false;
+	}
+
+	/* Fork a child process to execute the healthcheck command inside container */
+	pid_t pid = fork();
+	if (pid == -1) {
+		nwarnf("Failed to fork process for healthcheck command: %s", strerror(errno));
+		close(stderr_pipe[0]);
+		close(stderr_pipe[1]);
+		return false;
+	}
+
+	if (pid == 0) {
+		/* Child process - execute the healthcheck command inside container */
+		close(stderr_pipe[0]); /* Close read end of stderr pipe */
+
+		/* Redirect stdout to /dev/null and stderr to pipe */
+		int devnull = open("/dev/null", O_WRONLY);
+		if (devnull != -1) {
+			dup2(devnull, STDOUT_FILENO);
+			close(devnull);
+		}
+		dup2(stderr_pipe[1], STDERR_FILENO); /* Redirect stderr to pipe */
+		close(stderr_pipe[1]);
+
+		/* Build runtime command for direct execution */
+		/* Format: runtime exec container_id command args... */
+		char **runtime_argv;
+
+		/* Count arguments needed */
+		int argc = 0;
+		while (config->test[argc] != NULL) {
+			argc++;
+		}
+
+		/* Allocate runtime command array: runtime + exec + container_id + command + args + NULL */
+		runtime_argv = calloc(3 + argc + 1, sizeof(char *));
+		if (runtime_argv == NULL) {
+			nwarn("Failed to allocate memory for runtime command");
+			_exit(127);
+		}
+
+		runtime_argv[0] = (char *)runtime_path; /* Runtime executable */
+		runtime_argv[1] = "exec";		/* Runtime subcommand */
+		runtime_argv[2] = (char *)container_id; /* Container ID */
+
+		/* Copy healthcheck command and arguments */
+		for (int i = 0; i < argc; i++) {
+			runtime_argv[3 + i] = config->test[i];
+		}
+		runtime_argv[3 + argc] = NULL; /* NULL terminator */
+
+		/* Execute the runtime command */
+		if (execvp(runtime_path, runtime_argv) == -1) {
+			/* If execvp fails, exit with error code */
+			_exit(127); /* Command not found */
+		}
+	} else {
+		/* Parent process - wait for child to complete */
+		close(stderr_pipe[1]); /* Close write end of stderr pipe */
+		int status;
+		pid_t wait_result = waitpid(pid, &status, 0);
```

Comment:
The `healthcheck_execute_command()` function doesn't enforce the configured timeout parameter. The `waitpid(pid, &status, 0) which waits indefinitely, making the timeout parameter non-functional.